ktor-server/ktor-server-plugins/ktor-server-single-page/jvm/src/io/ktor/plugins/SinglePageApplication.kt

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.plugins
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.http.content.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.ktor.util.*
+import io.ktor.util.pipeline.*
+import java.io.*
+
+/**
+ * This plugin provides a single page application
+ */
+public class SinglePageApplication internal constructor(configuration: Configuration) {
+
+    internal val defaultPage: String = configuration.defaultPage
+
+    internal val applicationRoute: String = configuration.applicationRoute
+
+    internal val filesPath: String = configuration.filesPath
+
+    internal val ignoredFiles: MutableList<(String) -> Boolean> = configuration.ignoredFiles
+
+    internal val usePackageNames: Boolean = configuration.useResources
+
+    /**
+     * Configuration type for the [SinglePageApplication] plugin
+     */
+    public class Configuration(
+        /**
+         * Default name of file or resource to serve when [applicationRoute] is requested
+         */
+        public var defaultPage: String = "index.html",
+
+        /**
+         * The URL path under which the content should be served
+         */
+        public var applicationRoute: String = "/",
+
+        /**
+         * Path under which the static content is located.
+         * Corresponds to the folder path if the [useResources] is true, resource path otherwise
+         */
+        public var filesPath: String = "",
+
+        /**
+         * Specifies if static content is a resource or folder
+         */
+        public var useResources: Boolean = false,
+
+        /**
+         * List of callbacks checking if file or resource in [filesPath] is ignored
+         * Request for such files or resources fails with 404 Forbidden status
+         */
+        internal val ignoredFiles: MutableList<(String) -> Boolean> = mutableListOf()
+    ) {
+        /**
+         * Register a [block] in [ignoredFiles]
+         * [block] returns true if [path] should be ignored.
+         */
+        public fun ignoreFiles(block: (path: String) -> Boolean) {
+            ignoredFiles += block
+        }
+
+        /**
+         * Creates a single page plugin application configuration for the Angular project.
+         * Resources will be shared from filesPath directory. Root file is index.html
```

Comment:
I'd remove 'single page' to shorten the description (add added some articles):

> Creates an application configuration for the Angular project. Resources will be shared from the filesPath directory. The root file is index.html

The same for other configurations